### PR TITLE
Remove ArcsLocal and merge ArcsEnvironment and DeviceClient.

### PR DESCRIPTION
**Pull request by yuangu-google**

### CODE REVIEW

File: javaharness/java/arcs/android/client/AndroidArcsClient.java [hunk 1]
```diff
@@ -1,37 +1,79 @@
...
-import arcs.api.Arcs;
 import arcs.api.PECInnerPort;
 import arcs.api.PecPortManager;
 import arcs.api.PortableJsonParser;
 import arcs.api.UiBroker;
 
-// This class implements Arcs API for callers running in Android service
-// different that the one hosting the Arcs Runtime.
-public class ArcsAndroid implements Arcs {
+// This class implements Constants API for callers running in Android service
```
Comment (mmandlis) [inline javaharness/java/arcs/android/client/AndroidArcsClient.java]: what is "Constants API"?

File: javaharness/java/arcs/android/client/ArcsServiceBridge.java [hunk 2]
```diff
@@ -71,13 +70,12 @@ public void registerRenderer(String modality, IRemoteOutputCallback callback) {
...
   }
 
-  @Override
   public void sendMessageToArcs(String message) {
     runServiceMethod(service -> service.sendMessageToArcs(message));
   }
 
   /**
-   * Connects to the Arcs service (if not already connected), and then runs the given service
+   * Connects to the Constants service (if not already connected), and then runs the given service
```
Comment (mmandlis) [inline javaharness/java/arcs/android/client/ArcsServiceBridge.java]: same question - what is "Constants service"?
all i see with the name `Constants` is an interface with constants only, but it appears a lot in comments and logs, now sure i understand.
Also I don't think this is a good name - "Arcs API" and "Arcs service" seems clearer to me.

File: javaharness/java/arcs/android/demo/ui/AutofillDemoActivity.java [hunk 3]
```diff
@@ -86,6 +87,6 @@ private void capturePerson() {
     CapturePerson capturePerson = new CapturePerson();
     capturePerson.setId("capture-person-particle");
     capturePerson.setJsonParser(jsonParser);
-    arcs.runArc("IngestPeople", "capture-person-arc", "capture-person-pec", capturePerson);
+    constants.runArc("IngestPeople", "capture-person-arc", "capture-person-pec", capturePerson);
```
Comment (mmandlis) [inline javaharness/java/arcs/android/demo/ui/AutofillDemoActivity.java]: i don't see `runArc` method in `Constants` interface, how does this work?

File: javaharness/java/arcs/android/demo/ui/BUILD [hunk 4]
```diff
@@ -10,14 +10,14 @@ load("@build_bazel_rules_android//android:rules.bzl", "android_library")
...
+    javacopts = ["-Xep:AndroidJdkLibsChecker:OFF"],
     manifest = "AndroidManifest.xml",
     resource_files = glob(["res/**"]),
-    javacopts = ["-Xep:AndroidJdkLibsChecker:OFF"],
     deps = [
         "//javaharness/java/arcs/android/api",
-        "//javaharness/java/arcs/android/impl",
         "//javaharness/java/arcs/android/client",
         "//javaharness/java/arcs/android/demo/service",
+        "//javaharness/java/arcs/android/impl",
```
Comment (mmandlis) [inline javaharness/java/arcs/android/demo/ui/BUILD]: i patched the PR and tried blaze build and got 
```
ERROR: /Users/mmandlis/arcs/arcs-1/javaharness/java/arcs/android/demo/ui/BUILD:10:1: in android_library rule //javaharness/java/arcs/android/demo/ui:ui: target '//javaharness/java/arcs/android/demo/service:service' is not visible from target '//javaharness/java/arcs/android/demo/ui:ui'. Check the visibility declaration of the former target if you think the dependency is legitimate
```

File: javaharness/java/arcs/android/impl/ArcsLocal.java [hunk 5]
```diff
@@ -1,102 +0,0 @@
...
-        }
-        pecInnerPort.mapParticle(particleData.getParticle());
-      }
-    }
-    environment.sendMessageToArcs(constructRunArcRequest(arcData));
-  }
-
-  @Override
-  public void stopArc(ArcData arcData) {
-    environment.sendMessageToArcs(constructStopArcRequest(arcData));
```
Comment (mmandlis) [inline javaharness/java/arcs/android/impl/ArcsLocal.java]: copied from comment in original pr:

This class was useful for cases when you wanted to run an Arc from within the Arcs service (this is how the "ingestion-arc" would be started, it just wasn't part of the prototype, but it is part of the design).
I think it's nice if you could use the same interface to run an Arc from any service - Arcs or other Android service, this is why I created the Arcs interface and ArcsLocal & ArcsAndroid implementations (the names might be changed, but is there a reason to get rid of ArcsLocal implementation as a concept?)

File: javaharness/java/arcs/android/client/ArcsServiceBridge.java (hunk 2 above)
Comment (cromwellian) [reply javaharness/java/arcs/android/client/ArcsServiceBridge.java]: A common Java idiom is to make an interface with a bunch of constants, and then "implement" it to import them into your class, perhaps that's what it is.

**Pull request closed without merging**